AutoCollection/HeartBeat.ts

```diff
@@ -61,16 +70,79 @@ class HeartBeat {
             properties["appSrv_wsHost"] = process.env.WEBSITE_HOSTNAME || "";
         } else if (process.env.FUNCTIONS_WORKER_RUNTIME) { // Function apps
             properties["azfunction_appId"] = process.env.WEBSITE_HOSTNAME;
-        } 
-        this._client.trackMetric({name: Constants.HeartBeatMetricName, value: 0, properties: properties});
-        callback();
+        } else if (config && this._isVM === undefined) {
+            waiting = true;
+            this._getAzureComputeMetadata(config, () => {
+                if (this._isVM && Object.keys(this._vmData).length > 0) { // VM
+                    properties["azInst_vmId"] = this._vmData["vmId"] || "";
+                    properties["azInst_subscriptionId"] = this._vmData["subscriptionId"] || "";
+                    properties["azInst_osType"] = this._vmData["osType"] || "";
+                }
+                this._client.trackMetric({name: Constants.HeartBeatMetricName, value: 0, properties: properties});
+                callback();
+            });
```

Comment:
`trackMetric` here will only get called the first time. You should call `_getAzureComputeMetadata` only once ever, but trackMetric should still be called at the interval

Reply:
@markwolff I see, that makes sense. Do you think it makes sense to write `_isVM`, and other properties `azInst_vmId`, `azInst_subscriptionId`, and `azInst_osType` to environment variable?

Reply:
I think you can just write them to `this`

Reply:
oh! I misunderstood. I thought heartmetric instance is created each time... I see.. will update thanks.